### PR TITLE
feat(TAD-6145): init Linechart component

### PR DESCRIPTION
**Pull request by cbaune-talend**
**What is the problem this PR is trying to solve?**
Currently the dataviz package does not include any linegraph

**What is the chosen solution to this problem?**
Add a linegraph component in the dataviz package

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/dataviz/src/components/LineChart/LineChart.component.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		legend,
+	} = chartOptions;
+
+	return (
+		<ResponsiveContainer width={width || "100%"} height={height || "100%"} debounce={1}>
+		  <RLineChart
+			data={data}
+			margin={margin || {}}
+		  >
+			{ !!showGridLines && <CartesianGrid stroke="#F2F2F2" strokeDasharray="2" vertical={false} />}
```
Comment (Nigui) [inline packages/dataviz/src/components/LineChart/LineChart.component.tsx:33]: Could use one of those [gray values](https://talend.surge.sh/design-system/?path=/docs/design-tokens-design-tokens--page&globals=theme:light#grayscale) or use directly design token

File: packages/dataviz/src/components/LineChart/LineChart.types.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	height?: string,
+	margin?: {
+		top: number,
+		right: number,
+		bottom: number,
+		left: number,
+	}
+	tooltip?: {
+		custom?: boolean,
+		constentStyle?: any,
```
Comment (Nigui) [inline packages/dataviz/src/components/LineChart/LineChart.types.ts:27]: typing error ? `contentStyle` ?

File: packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	},
+}
+
+export const CustomTooltip = ({ active, payload, label, external }: LineChartTooltipProps) => {
+	const { chartOptions, linesConfig } = external;
+
+	const getLineUnit = (axis: 'left' | 'right' | undefined) => axis === 'right' ? chartOptions.rightYAxisOptions?.rechartsOptions?.unit : chartOptions.leftYAxisOptions?.rechartsOptions?.unit ;
+
+	const getLineIconBackground = (strokeDasharray: string | number | undefined, color: string) => {
+		console.log('stroke : ', strokeDasharray);
```
Comment (Nigui) [inline packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:24]: to remove

**Review by Nigui — COMMENTED** [review]
Good start imo !
You could use design tokens for size and colors

File: packages/dataviz/src/components/LineChart/LineChart.component.tsx (hunk 1 above)
Comment (Benrajalu) [reply packages/dataviz/src/components/LineChart/LineChart.component.tsx:33]: Those gray values are indeed deprecated. Sorry, we'll remove that part of the doc soon. The tokens are the source of truth: https://talend.surge.sh/design-tokens/?path=/docs/welcome--page

File: packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		return color;
+	};
+
+	if (active && payload && payload.length) {
+	  return (
+		<div className={classNames(styles['line-chart-custom-tooltip-wrapper'])}>
+			<p className={classNames(styles['line-chart-custom-tooltip-wrapper-title'])}>{label}</p>
+			{linesConfig.map(config => (
+		  		<p id={`tooltip_item_${config.key}`}>
+					<div
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:36]: using div inside a p is not allowed

**Comment by cbaune-talend** [comment]
Added some tests and more storybook examples.
Updated CSS to use design tokens.

So this is the first version of the feature.
A second version should comme after to simplify the API : taking some hard choices on the graph look&feel to greatly simplify the props of the component.

File: packages/dataviz/src/components/LineChart/LineChart.scss [hunk 5]
```diff
@@ -0,0 +1,72 @@
+@use '~@talend/design-tokens/lib/_tokens.scss' as tokens;
+
+.line-chart-custom-tooltip-wrapper {
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChart.scss:3]: please use sass & BEM in this module

File: packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+						className={classNames(
+							styles['line-chart-custom-tooltip-line-icon'],
+							)}
+						style={{ background: getLineIconBackground(config.rechartsOptions?.strokeDasharray, config.color)}}
+					/>
+					<span className={classNames(styles['line-chart-custom-tooltip-line-label'])}>
+						{ config.tooltipLabel ? config.tooltipLabel : config.key }
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:43]:
```suggestion
						{ config.tooltipLabel ?? config.key }
```

File: packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	const getLineUnit = (axis: 'left' | 'right' | undefined) => axis === 'right' ? chartOptions.rightYAxisOptions?.rechartsOptions?.unit : chartOptions.leftYAxisOptions?.rechartsOptions?.unit ;
+
+	const getLineIconBackground = (strokeDasharray: string | number | undefined, color: string) => {
+		if(strokeDasharray) {
+			return `repeating-linear-gradient(to right, ${color} 0, ${color} 10px,transparent 10px,transparent 12px)`;
+		}
+		return color;
+	};
+
+	if (active && payload && payload.length) {
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:30]:
```suggestion
	if (active && payload?.length) {
```

File: packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+	const { chartOptions, linesConfig } = external;
+
+	const getLineUnit = (axis: 'left' | 'right' | undefined) => axis === 'right' ? chartOptions.rightYAxisOptions?.rechartsOptions?.unit : chartOptions.leftYAxisOptions?.rechartsOptions?.unit ;
+
+	const getLineIconBackground = (strokeDasharray: string | number | undefined, color: string) => {
+		if(strokeDasharray) {
+			return `repeating-linear-gradient(to right, ${color} 0, ${color} 10px,transparent 10px,transparent 12px)`;
+		}
+		return color;
+	};
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChartTooltip.component.tsx:28]: this should be outside the component

File: packages/dataviz/src/components/LineChart/LineChartLegend.component.tsx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+	const getLineIconBackground = (strokeDasharray: string | number | undefined, color: string) => {
+		if(strokeDasharray) {
+			return `repeating-linear-gradient(to right, ${color} 0, ${color} 10px,transparent 10px,transparent 12px)`;
+		}
+		return color;
+
+	};
+
+	if (payload && payload.length) {
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChartLegend.component.tsx:47]:
```suggestion
	if (payload?.length) {
```

**Review by gjou-tlnd — CHANGES_REQUESTED** [review]

File: packages/dataviz/src/components/LineChart/LineChart.scss [hunk 10]
```diff
@@ -1,72 +1,79 @@
...
-		color: tokens.$coral-color-neutral-text;
+		&__line-label {
+			font: tokens.$coral-paragraph-m;
+			color: tokens.$coral-color-neutral-text;
+		}
+
+		&__line-value {
+			font: tokens.$coral-paragraph-m-bold;
+			color: tokens.$coral-color-neutral-text;
+			margin-left: tokens.$coral-size-xxs;
```
Comment (Benrajalu) [inline packages/dataviz/src/components/LineChart/LineChart.scss:41]:
```suggestion
			margin-left: tokens.$coral-spacing-xxs;
```

File: packages/dataviz/src/components/LineChart/LineChart.scss [hunk 11]
```diff
@@ -1,72 +1,79 @@
...
-	// box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.16);
-	box-shadow: tokens.$coral-elevation-shadow-neutral-m;
-	border-radius: tokens.$coral-radius-s;
+.#{$module} {
+	&-custom-tooltip {
+		display: flex;
+		flex-direction: column;
+		justify-content: center;
+		align-items: flex-start;
+		padding: tokens.$coral-size-m;
```
Comment (Benrajalu) [inline packages/dataviz/src/components/LineChart/LineChart.scss:11]:
```suggestion
		padding: tokens.$coral-spacing-m;
```

File: packages/dataviz/src/components/LineChart/LineChart.scss [hunk 12]
```diff
@@ -1,72 +1,79 @@
...
-	justify-content: flex-start;
-	align-items: center;
-	gap: tokens.$coral-size-m;
-	list-style: none;
+	&-custom-legend {
+		display: flex;
+		flex-direction: row;
+		justify-content: flex-start;
+		align-items: center;
+		gap: tokens.$coral-size-m;
```
Comment (Benrajalu) [inline packages/dataviz/src/components/LineChart/LineChart.scss:55]:
```suggestion
		gap: tokens.$coral-spacing-m;
```

**Review by jmfrancois — CHANGES_REQUESTED** [review]
please remove the package-lock

File: packages/dataviz/src/components/LineChart/LineChart.component.test.tsx [hunk 13]
```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-console */
+import React from 'react';
+import { mount } from 'enzyme';
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChart.component.test.tsx:3]: IMHO, new code should use RTL, any thoughts @jsomsanith ?

File: packages/dataviz/src/components/LineChart/LineChart.component.test.tsx [hunk 14]
```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-console */
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChart.component.test.tsx:1]:
```suggestion
```

File: packages/dataviz/src/components/LineChart/LineChart.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,317 @@
+import React from 'react';
+import { Meta, Story } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import  LineChart, { LineChartProps } from './LineChart.component';
+
+const Template: Story<LineChartProps> = args => <LineChart {...args} />;
```
Comment (gjou-tlnd) [inline packages/dataviz/src/components/LineChart/LineChart.stories.tsx:6]:
```suggestion
```

you can use the new CSF syntax:
```suggestion
export const FullyCustomisedLineChart = Template.bind({});```
```suggestion
const Template: Story<LineChartProps> = args => <LineChart {...args} />;
```

```
export const FullyCustomisedLineChart = {
   args: {...}
};
```

**Comment by gjou-tlnd** [comment]
Tooltip style doesn't match bar/geo charts, is this expected? Can't we use only one style?
Screenshot 2022-04-11 at 14 18 07
Screenshot 2022-04-11 at 14 18 26

**Comment by gjou-tlnd** [comment]
> Tooltip style doesn't match bar/geo charts, is this expected? Can't we use only one style? Screenshot 2022-04-11 at 14 18 07 Screenshot 2022-04-11 at 14 18 26



> > Tooltip style doesn't match bar/geo charts, is this expected? Can't we use only one style?
> 
> If we agree on the white tooltip for all chart, we can go for it here 👍🏼

I'll let @cbaune-talend lead the subject with product designers to make sure everyone is on board

**Comment by cbaune-talend** [comment]
Last update ! Taking into account the comments from @gjou-tlnd & @Benrajalu : 
- reworked the tooltip to stick to DS principles
- switched to RTL for testing
- use new CSF syntax for stories

**Comment by gjou-tlnd** [comment]
> Last update ! Taking into account the comments from @gjou-tlnd & @Benrajalu :
> 
> * reworked the tooltip to stick to DS principles
> * switched to RTL for testing
> * use new CSF syntax for stories
> 
> image

We still end up with two tooltip implementations, please merge them

**Review by Nigui — CHANGES_REQUESTED** [review]
Please export LineChart components and types from dataviz module

**Comment by Nigui** [comment]
> Please export LineChart components and types from dataviz module

You can apply this patch to do it 
(replace `.txt` by `.patch` before calling `> git am 0001-feat-TDC-6145-export-LineChart-component-and-types.patch` command)

[0001-feat-TDC-6145-export-LineChart-component-and-types.patch](https://github.com/Talend/ui/files/8539584/0001-feat-TDC-6145-export-LineChart-component-and-types.txt)

**Comment by gjou-tlnd** [comment]
> @Nigui added your commit to export the component.
> 
> @gjou-tlnd : since the two tooltip component have different needs, I don't think it would be wise to merge them in a single tooltip component. None the less, I modified the `TooltipContent` css to use the same DS tokens, so the two looks the same. If we really want to make a unique tooltip component for all the graph we could create another ticket and discuss a feasible future solution.

I was thinking of the creating a container only responsible for color/border/padding and using children for actual content... but using DS tokens is a good first step, thks

**Review by Nigui — APPROVED** [review]

**Review by jmfrancois — APPROVED** [review]
animation of the LineChart seems really slow compare to other charts.

**Comment by cbaune-talend** [comment]
> animation of the LineChart seems really slow compare to other charts.

fixed in [3fb5955](https://github.com/Talend/ui/pull/3785/commits/3fb595578aae24782bab2564c1b85fb2967532c7)

**Review by gjou-tlnd — APPROVED** [review]

**Pull request merged**